### ISSUE TITLE
[3.4] Images in large thumbnails and inline editor previews are cropped.

### ISSUE THREAD

**Issue by aelvan** [craftcms/cms#5515]
### Description

In versions prior to 3.4, images were fitted into the large thumbnail containers and the preview area of the inline editor. In 3.4 they're cropped, making it impossible for users to know what the original format of the image is, when picking an asset. 

In Craft 3.3:
![image](https://user-images.githubusercontent.com/2675644/73298560-d132d900-420d-11ea-9e7e-9ba194e097e2.png)

In Craft 3.4:
![image](https://user-images.githubusercontent.com/2675644/73298604-e871c680-420d-11ea-9993-ce37b6a55b72.png)

The only way to see the actual format of the image in 3.4, is by opening the full image editor, or going to the edit asset page and clicking preview.

Although this makes the images look better in the said contexts, I think this is a major step-back AX-wise. It's essential for the author to know the original and intended format of an image, to make good calls on which image to use when authoring content.

Please make it fit!

### Steps to reproduce

1. Upload two images, one landscape and one portrait.
2. Go to an asset element index and toggle to large thumbnail view. 

### Additional info

- Craft version: 3.4.0
- PHP version: PHP 7.3.8
- Database driver & version: Mysql 5.7.26

**Comment by RitterKnightCreative** [comment]
I think a case can be made for having both options in the CMS, show original aspect ratio and the cropped version, with the ability to toggle between them.

Seeing as an image can often live as multiple different sizes/transforms when finally rendered on the site, the cropped version is nice because you can see where Craft will center it up for the transform.  That's more useful in some cases, less so in others.

If you have a portrait shot of people, for example,  it's a quick indication that you also need to update the focus point or you might marie antoinette someone. ;)

Similar to #5518...

**Comment by aelvan** [comment]
@RitterKnightCreative I agree that it's a good feature request to be able to see how the cropped version of an image would look. But if this change was meant to solve that, I think it's a pretty bad solution. It only partly solves the problem, seeing an image cropped to a square is only barely useful if the final image is suppose to be cropped for 16:9. And removing the option for authors to see the original aspect ratio of the image, as intended by the photographer/art director, is a massive downside.

What would be a better solution would be to tie the cropping to the Asset field through settings. If it was possible to define a custom crop ratio (along with "no crop", ie "fit") in an Asset field's settings, and that crop showed up in the thumbnails on the field when you'd selected it, that'd be pretty amazing. And, if "fit" was used in the asset selection modals and the assets element index, everyone would be happy.

**Comment by matthillco** [comment]
I find this to be a backwards step in Craft's image asset workflow. I work with a lot of mixed ratio images and I need to be able to see at a glance the ratio of an image before selecting it. I get that cropped images make for attractive grids, but in practice they aren't that useful for content-editors.

It would be great if Craft could provide a way to configure this, otherwise it's going to be pretty painful to use this going forward.

**Comment by aelvan** [comment]
I created a small plugin which fixes this while we wait for 3.5: 
https://github.com/aelvan/more-fit-images

**Comment by ryanmasuga** [comment]
We do _a lot_ of work on numerous client sites that have 1000's of images that are imported into Craft every day and use all sorts of aspect ratios (social media images, broadcast images). Upgrading any of those sites to Craft 3.4 would be a disaster _because of these uniformly square cropped images alone_. The editors would be helpless choosing from 20+ similar thumbnails, looking for a certain 16:9, for example.

**Comment by brandonkelly** [comment]
This is now addressed for Craft 3.5.

<img width="548" alt="Asset thumbnails revealing their true aspect ratios" src="https://user-images.githubusercontent.com/47792/76114408-da486000-5f9a-11ea-9da3-ee2dde3e8cea.png">

<img width="396" alt="An asset HUD with a thumbnail revealing the image’s aspect ratio" src="https://user-images.githubusercontent.com/47792/76114475-019f2d00-5f9b-11ea-9936-c06da821d13c.png">

The table view will continue using circular crops, though, as I think with such a small image, being able to make out some detail of the image is more important than knowing the aspect ratio.

**Comment by brandonkelly** [comment]
I changed my mind on how we should be handling small thumbnails. Showing them at the correct aspect ratio isn’t really the end of the world. So I was able to implement the change for 3.4 with significantly less work than what I had done on the 3.5 branch. Will be in the next 3.4 release.

To get the fix early, change your `craftcms/cms` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#8f6e9e8945d4df9627f0237dd47b0e66e4d52541 as 3.4.20",
  "...": "..."
}
```

Or if you’re already running Craft 3.5 Beta, then change your `craftcms/cms` requirement to `"3.5.x-dev"`.

Then run `composer update`.
